docs/quickstart/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,890 @@
+# Stateful Precompile Generation Tutorial
+
+In this tutorial, we are going to walkthrough how we can generate a stateful precompile from scratch. Before we start, let's brush up on what a precompile is, what a stateful precompile is, and why this is extremely useful.
+
+## Background
+
+### Precompiled Contracts
+
+Ethereum uses precompiles to efficiently implement cryptographic primitives within the EVM instead of re-implementing the same primitives in Solidity. The following precompiles are currently included: ecrecover, sha256, blake2f, ripemd-160, Bn256Add, Bn256Mul, Bn256Pairing, the identity function, and modular exponentiation.
+
+We can see these precompile mappings from address to function here in the Ethereum VM.
+
+``` go
+// PrecompiledContractsBerlin contains the default set of pre-compiled Ethereum
+// contracts used in the Berlin release.
+var PrecompiledContractsBerlin = map[common.Address]PrecompiledContract{
+	common.BytesToAddress([]byte{1}): &ecrecover{},
+	common.BytesToAddress([]byte{2}): &sha256hash{},
+	common.BytesToAddress([]byte{3}): &ripemd160hash{},
+	common.BytesToAddress([]byte{4}): &dataCopy{},
+	common.BytesToAddress([]byte{5}): &bigModExp{eip2565: true},
+	common.BytesToAddress([]byte{6}): &bn256AddIstanbul{},
+	common.BytesToAddress([]byte{7}): &bn256ScalarMulIstanbul{},
+	common.BytesToAddress([]byte{8}): &bn256PairingIstanbul{},
+	common.BytesToAddress([]byte{9}): &blake2F{},
```

Comment:
There is no this function at https://github.com/ethereum/go-ethereum/blob/master/core/vm/contracts.go#L52. 

* Can you link to the source code in this doc?
* Please make sure they match to the ethereum code.

Reply:
I'm referencing berlin upgrade. https://github.com/ethereum/go-ethereum/blob/master/core/vm/contracts.go#L81

Reply:
Then please hyperlink to the code.